### PR TITLE
Discharge calculation

### PR DESCRIPTION
**Pull request by KristianNelson**
Calculate the waste discharge flow through treatment plants (million gallons per day) per person.

### CODE REVIEW

File: R/count_watershed_data.R [hunk 1]
```diff
@@ -221,9 +221,30 @@ count_watershed_data <- function(data_dir,
...
+                          "lon",
+                          "lat",
+                          "DISCHARGE_METHOD",
+                          "EXIST_TOTAL",
+                          "PROJ_TOTAL",
+                          "PRES_RES_RECEIVING_COLLCTN",
+                          "PRES_RES_ONSITE_WTS"
+          )) -> wasteflow_select
 
+        wasteflow_select[is.na(wasteflow_select)] <- 0
```
Comment (swd-turner) [inline R/count_watershed_data.R:239]: best to keep `NA` as `NA`. Instead just add the argument `na.rm = T` to the sums below.

File: R/count_watershed_data.R [hunk 2]
```diff
@@ -221,9 +221,30 @@ count_watershed_data <- function(data_dir,
...
 
+        sum(wasteflow_select$EXIST_TOTAL) -> totalflow_thru_MGD
+
+        sum(wasteflow_select$PRES_RES_RECEIVING_COLLCTN) -> total_pop_served
```
Comment (swd-turner) [inline R/count_watershed_data.R:243]: We want to make sure that the population served is calculated only for rows that have a discharge value. So I suggest something like:
```
wasteflow_select %>%
   filter(!is.na(EXIST_TOTAL) & !is.na(PRES_RES_RECEIVING_COLLCTN)) -> wasteflow_select_complete

wasteflow_select_complete["EXIST_TOTAL"] %>% sum() -> totalflow_thru_MGD
wasteflow_select_complete["PRES_RES_RECEIVING_COLLCTN"] %>% sum() -> total_pop_served
```

File: R/count_watershed_data.R [hunk 3]
```diff
@@ -431,7 +452,8 @@ count_watershed_data <- function(data_dir,
               "runoff from developed land",      "%",        developed_runoff_percent,
               "total irrigation consumption",    "BCM",      total_irr_bcm,
               "population",                      "people",   wtrshd_population,
-              "population water consumption",    "ltr/sqkm", pop_water_consum
+              "population water consumption",    "ltr/sqkm", pop_water_consum,
+              "waste discharge per person",      "MGD/pers", wastedischarge_per_person
```
Comment (swd-turner) [inline R/count_watershed_data.R:456]: Please convert from MGD to cumecs (m3/s) for consistency with the rest of the package, which is using metric units.

File: R/count_teleconnections.R [hunk 4]
```diff
@@ -321,6 +321,11 @@ count_watershed_teleconnections <- function(data_dir,
             .[["Reclass"]] %>% unique()
         }) %>% unlist() %>% unique() %>% length() -> n_economic_sectors
 
+        # waste discharge per person
+        map(city_watershed_data, function(x){
+          x$metrics %>% filter(metric == "waste discharge per person") %>%
+            .[["value"]]
+        }) %>% unlist() %>% sum() -> wasteMGD_per_person
```
Comment (swd-turner) [inline R/count_teleconnections.R:328]: Please check this sum. Are you summing discharge per person across watersheds?

Comment (swd-turner) [reply R/count_teleconnections.R:328]: Appropriate method here would be to sum the discharge across watersheds, sum the associated population across watersheds and then compute the per person average here.

Comment (KristianNelson) [reply R/count_teleconnections.R:328]: You are absolutely right on that. I'll make sure to do the calculations after those are summed up!

**Comment by KristianNelson** [comment]
@swd-turner When this clears, please pull down and try to run on your machine. I am able to run single or a couple cities at a time, but when I try to do the full run it gives me an error saying it cannot allocate a file the size of 1.1 GB. I want to see if this is just my machine or if its an error in the code. Thanks!

**Comment by KristianNelson** [comment]
@swd-turner This latest push just makes sure that all features are in the same projection. When I made this change, the watershed population didn't change and is still up at 3 million. Before this change, we saw a watershed population of 60000 for Norfolk VA, and this is now coming in at around 1 million (which seems like a more reasonable number).

**Comment by KristianNelson** [comment]
Numbers are better but still looking a little off. Abilene and Odessa are still showing human water consumption/discharge as accounting for 200% of the runoff during the driest month.

File: R/constants.R [hunk 5]
```diff
@@ -20,4 +20,14 @@ day_to_sec <- 86400
...
+
+MGD_to_m3sec <- 0.0438
+
+BCMyr_to_m3sec <- 31.71
+
+BCMmonth_to_m3sec <- 373.36
+
+ltrday_to_m3sec <- 1.15741e-8
+
+onekmsq_to_60msq <- 2.78
```
Comment (swd-turner) [inline R/constants.R:33]: can you double check this?... 60m by 60m = 3600m2... then 1km2 is 1000 * 1000 = 1,000,000 m2.

1,000,000 / 3600 = 277.78    (maybe I've misunderstood the resolution of the raster?)

Comment (KristianNelson) [reply R/constants.R:33]: Nope you are correct that is my bad, I was just off by 2 decimal places. Doe everything else look correct?

File: R/constants.R [hunk 6]
```diff
@@ -20,4 +20,14 @@ day_to_sec <- 86400
 
 missing_watersheds <- c(3589, 3850, 3678, 3582, 3615, 4226, 4227, 2336, 3448, 2162, 3474) # for runoff
 
-avg_wateruse_ltr <- 333
+avg_wateruse_ltr <- 200
```
Comment (swd-turner) [inline R/constants.R:23]: `avgwateruse_ltr_per_day`

File: R/count_watershed_data.R [hunk 7]
```diff
@@ -221,9 +225,46 @@ count_watershed_data <- function(data_dir,
 
         #-------------------------------------------------------
         # TELECONNECTION - WASTE WATER TREATMENT PLANTS
-        wasteflow_points[watersheds_select, ] -> wasteflow_select
+        wasteflow_points[watersheds_select, ] %>%
```
Comment (swd-turner) [inline R/count_watershed_data.R:228]: This is no longer used, right? Best to remove redundancy in the code.

Comment (KristianNelson) [reply R/count_watershed_data.R:228]: Will do, I just wanted to make sure we liked the results of the new calculation before I got rid of it!

**Comment by KristianNelson** [comment]
@swd-turner I figured out the issue with the population. I was using the get_zonal_data() function incorrectly. Fixing it now and it should be more accurate using James' method in his email.

**Comment by KristianNelson** [comment]
I'm going to need some help getting this to clear... No problems on my machine so I am not too sure what I need to be looking for. From the build info it looks like it has something to do with `rgdal` but I went through all of that and made sure everything was correct.

**Pull request merged**